oap-server/analyzer/agent-analyzer/src/main/java/org/apache/skywalking/oap/server/analyzer/provider/trace/parser/listener/SegmentAnalysisListener.java

```diff
@@ -170,14 +170,17 @@ private void appendSearchableTags(SpanObject span) {
 
     @Override
     public void build() {
-        if (log.isDebugEnabled()) {
-            log.debug("segment listener build, segment id: {}", segment.getSegmentId());
-        }
-
         if (sampleStatus.equals(SAMPLE_STATUS.IGNORE)) {
+            if (log.isDebugEnabled()) {
+                log.debug("segment ignored, trace id: {}", segment.getTraceId());
```

Comment:
Does not-sampled segment have trace id?

Reply:
yes, there's always trace id.

Reply:
I need to recheck the codes, could you point out how this happens? Having trace id but no segment id.

Reply:
In addition to lowering the sample rate, everything else configuration can be configured with the default value in master branch.

Reply:
Is this caused by backend sampling? I remember all reported data should all have trace id and segment id. That is what I am not following. What the missing logic here.

Reply:
Yes, I forgot to say it was back-end sampling. :-P

Reply:
I still need to find out why this happens. Still missing why segment id lost. 😥
Or just make logs more readable.

Reply:
<img width="648" alt="Screen Shot 2021-06-05 at 10 18 19" src="https://user-images.githubusercontent.com/15965696/120877213-d82ee000-c5e7-11eb-9b43-f10f4984d1a3.png">

This is because we defer the value of segment in `parseFirst`.

I think we should review the `parse{First,Entry}` methods and move those span-unrelated logics into `parseSegment`, such as `segment.setSegmentId(segmentObject.getTraceSegmentId()`

Reply:
OK, this makes sense more. If we want this log out, I am fine to merge this.